### PR TITLE
[DEV] Update to fiona 1.10b3

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Here is my attempt to fulfill the request in #225.  Perhaps there is a slicker way to merge in the changes from `main` but I just 
* wiped out the old `recipe` sudir in `dev` and copied over the new one from `main`
* copied over `conda-forge.yml` from `main`
* added `recipe/conda_build_config.yaml` to use a new `fiona_dev` label.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Here are the errors:
```

tests/test_drvsupport.py:29
  /tmp/tests/test_drvsupport.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_drvsupport.py:70
  /tmp/tests/test_drvsupport.py:70: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_drvsupport.py:102
  /tmp/tests/test_drvsupport.py:102: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_drvsupport.py:165
  /tmp/tests/test_drvsupport.py:165: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_drvsupport.py:222
  /tmp/tests/test_drvsupport.py:222: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_drvsupport.py:251
  /tmp/tests/test_drvsupport.py:251: PytestUnknownMarkWarning: Unknown pytest.mark.gdal - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gdal

tests/test_fio_load.py:126
  /tmp/tests/test_fio_load.py:126: PytestUnknownMarkWarning: Unknown pytest.mark.iconv - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.iconv

tests/test_unicode.py:117
  /tmp/tests/test_unicode.py:117: PytestUnknownMarkWarning: Unknown pytest.mark.iconv - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.iconv

tests/test_unicode.py:145
  /tmp/tests/test_unicode.py:145: PytestUnknownMarkWarning: Unknown pytest.mark.iconv - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.iconv

tests/test_vfs.py:166
  /tmp/tests/test_vfs.py:166: PytestUnknownMarkWarning: Unknown pytest.mark.network - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.network

tests/test_vfs.py:176
  /tmp/tests/test_vfs.py:176: PytestUnknownMarkWarning: Unknown pytest.mark.network - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.network

tests/test_vfs.py:183
  /tmp/tests/test_vfs.py:183: PytestUnknownMarkWarning: Unknown pytest.mark.network - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.network

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
================ 16 deselected, 26 warnings, 4 errors in 1.85s =================
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,41 +1,52 @@
-{% set version = "1.8rc1" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:1]: I completely forgot we already had a branch for this!

**Review by ocefpaf — APPROVED** [review]

**Comment by sgillies** [comment]
@xylar right, test_perf.py was accidentally packaged in 1.10b3. aiohttp and fsspec are in requirements-dev.txt but not pyproject.toml, which I am correcting.

**Comment by xylar** [comment]
@sgillies, sounds good!

> aiohttp and fsspec are in requirements-dev.txt but not pyproject.toml, which I am correcting.

Oh, I only looked at the `pyproject.toml`.  Probably better to get rid of `requirements-dev.txt` to avoid confusion.

**Comment by xylar** [comment]
It also looks like `munch>=2.3.2` is in `requirements.txt` but not in `pyproject.toml`.  I've found in my own development that it's sometimes necessary to have redundant requirements files like this but it's sure error prone!

**Comment by xylar** [comment]
@sgillies, if you're happy with my latest changes, feel free to merge once CI passes.

**Comment by sgillies** [comment]
Ah, yeah, the munch requirement is a derelict. FWIW, the fiona requirements files exist only to support its CI.

File: recipe/0002-remove-test-perf.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+@@ -1,9 +0,0 @@
+-"""Performance test."""
+-
+-import fiona
+-
+-with fiona.open(
+-    "/app/ne_10m_admin_0_countries_lakes/ne_10m_admin_0_countries_lakes.shp"
+-) as src:
+-    with fiona.open("/tmp/test_perf.shp", "w", **src.profile) as dst:
+-        dst.writerecords(src)
```
Comment (sgillies) [inline recipe/0002-remove-test-perf.patch:13]: 👍

File: recipe/0001-remove-certifi-dependency.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+diff -ruN fiona-1.9.5/pyproject.toml fiona-1.9.5-patch/pyproject.toml
+--- fiona-1.9.5/pyproject.toml	2023-10-12 13:14:23.000000000 -0600
++++ fiona-1.9.5-patch/pyproject.toml	2023-11-28 10:47:25.666126879 -0700
+@@ -37,7 +37,6 @@
+ requires-python = ">=3.7"
+ dependencies = [
+     "attrs>=19.2.0",
+-    "certifi",
```
Comment (sgillies) [inline recipe/0001-remove-certifi-dependency.patch:8]: Fiona imports certifi, so this is needed.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,41 +1,53 @@
...
+    - gdal
+    - attrs >=19.2.0
+    - click >=8.0,<9.dev0
     - cligj >=0.5
     - click-plugins >=1.0
-    - six >=1.7
-    - munch
-    - enum34  # [py27]
+    - importlib-metadata  # [py<310]
+    - munch >=2.3.2
```
Comment (sgillies) [inline recipe/meta.yaml:48]: Can be removed, Fiona vendors munch now.

```suggestion
```

**Review by sgillies — COMMENTED** [review]
@xylar Thank you! I made several suggestions inline.

File: recipe/0001-remove-certifi-dependency.patch (hunk 3 above)
Comment (xylar) [reply recipe/0001-remove-certifi-dependency.patch:8]: See the discussion starting at https://github.com/conda-forge/fiona-feedstock/pull/220#issuecomment-1830275315
and in
https://github.com/conda-forge/fiona-feedstock/pull/188
The issue is that conda-forge only supports `certifi` for python versions where it's actually needed.  But your inclusion upstream of `certifi` for all python causes pip check to fail.  So I have to remove it manually from the `pyproject.toml` to allow `pip check` to pass on downstream packages.

This is actually the whole reason I got involved with maintaining Fiona in the first place.  I was trying to fix downstream `pip check`.

File: recipe/meta.yaml (hunk 4 above)
Comment (xylar) [reply recipe/meta.yaml]: Vendoring other packages isn't ideal but if that's how it is...

File: recipe/0001-remove-certifi-dependency.patch (hunk 3 above)
Comment (xylar) [reply recipe/0001-remove-certifi-dependency.patch:8]: I'll just try adding it back.  It seems like it *should* work for all python so I'm not sure what the previous issue was...

**Comment by xylar** [comment]
Sorry for the confusion about `certifi`.  As long as `pip check` passes without the patch, I'm happy...

**Comment by xylar** [comment]
@sgillies, CI has passed.  This is ready for you to take another look.

**Review by sgillies — APPROVED** [review]
@xylar looks good!

**Pull request merged**